### PR TITLE
Android YouTube

### PR DESCRIPTION
**Pull request by ionela-voinescu**
Modify YouTube benchmark to make use of the new android benchmark API and provide an example of an android YouTube benchmark test.

### CODE REVIEW

File: libs/utils/android/workloads/youtube.py [hunk 1]
```diff
@@ -42,31 +40,46 @@ def __init__(self, test_env):
...
+        :type exp_dir: str
+
+        :param video_url: Video URL to be played
+        :type video_url: str
+
+        :param duration_s: Play video for this required number of seconds
+        :type duration_s: int
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
```
Comment (derkling) [inline libs/utils/android/workloads/youtube.py]: Why only energy?

There are not limitations from the Workload's module standpoint, which now supprots ftrace/systrace and energy:
https://github.com/ARM-software/lisa/blob/master/libs/utils/android/workload.py#L90

BTW thanks for the documentation! ;-)

File: libs/utils/android/workloads/youtube.py [hunk 2]
```diff
@@ -42,31 +40,46 @@ def __init__(self, test_env):
...
 
         # Use the monkey tool to start YouTube without playing any video.
         # This allows to subsequently set the screen orientation to LANDSCAPE
         # and to reset the frame statistics.
-        System.monkey(self.target, self.package)
+        System.monkey(self._target, self.package)
 
         # Force screen in LANDSCAPE mode
-        Screen.set_orientation(self.target, portrait=False)
+        Screen.set_orientation(self._target, portrait=False)
```
Comment (derkling) [inline libs/utils/android/workloads/youtube.py]: I forgot in my jankbench example to tune the screen brightness, can you add it at least in these new worklaods?

File: libs/utils/android/workloads/youtube.py [hunk 3]
```diff
@@ -78,16 +91,16 @@ def run(self, out_dir, collect,
...
-        System.force_stop(self.target, self.package, clear=False)
+        System.force_stop(self._target, self.package, clear=False)
 
         # Go back to home screen
-        System.home(self.target)
+        System.home(self._target)
 
         # Switch back to screen auto rotation
-        Screen.set_orientation(self.target, auto=True)
+        Screen.set_orientation(self._target, auto=True)
```
Comment (derkling) [inline libs/utils/android/workloads/youtube.py]: And brightness must be reset here...

File: tests/benchmarks/android_youtube.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        # Tools required by the experiments
+        "tools"   : [ 'trace-cmd' ],
+
+		# Default results folder
+		"results_dir" : "AndroidYouTube",
+
+    }
+
+	# Android Workload to run
+    bm_name = 'YouTube'
```
Comment (derkling) [inline tests/benchmarks/android_youtube.py]: Is there some spaces issue here, before the comment?

File: tests/benchmarks/android_youtube.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+video_duration_s = 60
+
+governors = [
+    'performance',
+    'ondemand',
+    'interactive',
+    'sched',
+    'schedutil',
+    'powersave',
+]
```
Comment (derkling) [inline tests/benchmarks/android_youtube.py]: Perhaps we should think about adding a new benchmark command line parameter here:
   https://github.com/ARM-software/lisa/blob/master/libs/utils/android/benchmark.py#L78
which allows to specify at command line which governors we want to test.

Let's keep for another time, but what do you think?... most of the time maybe we do not want to test with really all these governors... do you think it should be possible/useful to expose such a tunables in Jenkins?

File: libs/utils/android/workloads/youtube.py [hunk 6]
```diff
@@ -102,5 +105,7 @@ def run(self, out_dir, video_url, video_duration_s, collect=''):
 
         # Switch back to screen auto rotation
         Screen.set_orientation(self._target, auto=True)
+        # Set brightness back to auto
+        Screen.set_brightness(self._target, auto=True)
```
Comment (derkling) [inline libs/utils/android/workloads/youtube.py]: Oh... here is the patch... just squash with the first one ;-)

File: libs/utils/android/workloads/youtube.py (hunk 1 above)
Comment (ionela-voinescu) [inline libs/utils/android/workloads/youtube.py]: Just omission due to hurry. I'll modify the others too as they will suffer from the copy-paste syndrome.

File: libs/utils/android/workloads/youtube.py (hunk 2 above)
Comment (ionela-voinescu) [inline libs/utils/android/workloads/youtube.py]: Squashed! :imp:

Comment (derkling) [inline libs/utils/android/workloads/youtube.py]: But down here I still see two patches... ?!?

File: tests/benchmarks/android_youtube.py (hunk 4 above)
Comment (ionela-voinescu) [inline tests/benchmarks/android_youtube.py]: Yes, it was seeing them alright with `:set tabstop=4 shiftwidth=4 expandtab`. It's fixed now.

File: libs/utils/android/workloads/youtube.py (hunk 2 above)
Comment (ionela-voinescu) [inline libs/utils/android/workloads/youtube.py]: It's done now! I posted the comment a short while before I got the chance to push the branch. You move fast!

**Pull request merged**